ethcore/src/state/mod.rs

```diff
@@ -204,10 +259,12 @@ impl State {
 	}
 
 	fn insert_cache(&self, address: &Address, account: AccountEntry) {
-		if let Some(ref mut snapshot) = self.snapshots.borrow_mut().last_mut() {
-			if !snapshot.contains_key(address) {
-				snapshot.insert(address.clone(), self.cache.borrow_mut().insert(address.clone(), account));
-				return;
+		if account.is_dirty() {
```

Comment:
would be nice to get a quick comment on this chunk of code - meaning is not immediately obvious...